File: Backend/DigitalStore.Domain/Entities/ContentImage.cs
using System;

namespace DigitalStore.Domain.Entities
{
    public class ContentImage
    {
        public int Id { get; set; }
        public string ImageUrl { get; set; }
        public int DisplayOrder { get; set; }
        public string? AltText { get; set; }
        public int EntityTypeId { get; set; }
        public int EntityId { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: education_app/lib/features/quiz/domain/entities/content_image.dart
import 'package:equatable/equatable.dart';

class ContentImage extends Equatable {
  final int id;
  final String imageUrl;
  final int displayOrder;
  final String? altText;
  final int imageTypeId;

  const ContentImage({
    required this.id,
    required this.imageUrl,
    required this.displayOrder,
    this.altText,
    this.imageTypeId = 6,
  });

  bool get isFullPage => imageTypeId == 1;

  @override
  List<Object?> get props => [id, imageUrl, displayOrder, altText, imageTypeId];
}
